### ISSUE TITLE
processing plugin after it is loaded

### ISSUE THREAD

**Issue by vinitm** [jackfranklin/gulp-load-plugins#116]
```javascript
var gulp = require('gulp'),
    plugins = require('gulp-load-plugins')({
        pattern: ['*'],
        rename: {
            'browser-sync': 'browserSync',
        }
    });

plugins.browserSync = plugins.browserSync.create();
```

Is this the right way of manually calling `plugins.browserSync = plugins.browserSync.create();` after loading plugins or there is some option provided for doing the same?

**Comment by jackfranklin** [comment]
This is the right and only way to do this.

I'm 50/50 on wanting an API to do this. How common is it that plugins need this? I don't want to add yet another config option if this is a very rare case but if enough people need it I'd be open to suggestions.

**Comment by vinitm** [comment]
```javascript
plugins.browserSync = plugins.browserSync.create();
plugins.reload = plugins.browserSync.reload;
plugins.underscorify = plugins.underscorify.transform({
    extensions: ['tpl']
});
plugins.babelify=babelify.configure({
  presets: ["es2015", "react"]
});
```

Above are some of the plugins that I have been using, and it seems it is quite common for plugins to ask for plugin-specific options for initializing.

And Browsersync and Babel are only some of the many popular plugins which have a good user base.

**Comment by jackfranklin** [comment]
OK - particularly given babelify then I can see the potential argument. What are your thoughts on a potential API?

**Comment by vinitm** [comment]
```javascript
gulpLoadPlugins({
  camelize: true,
  lazy: true,
  rename: {},
  parseFns: {
    plugin1NameAfterRename: function(pluginInstance) {
      var parsedPlugin = doSomethingWith(pluginInstance);
      return parsedPlugin;
    },
    plugin2NameAfterRename: function(pluginInstance) {},
    plugin3NameAfterRename: function(pluginInstance) {}
  }
});
```

`parseFns` option can be provided which takes a parse function for each plugin and returns a parsed plugin.

**Comment by vinitm** [comment]
That is exactly what I mean

**Comment by jackfranklin** [comment]
@vinitm how do you feel about working on a PR? :)

**Comment by vinitm** [comment]
sure, I will create a PR

**Comment by jackfranklin** [comment]
Closed as of #119
